tests/workspace/test_builder.py

```diff
@@ -340,9 +345,68 @@ def test_environment_variables(session):
     builder = WorkspaceBuilder(sconf=workspace)
     builder.build(session)
 
-    assert session.getenv("FOO") == "BAR"
+    assert session.getenv("FOO") == "SESSION"
+    assert session.getenv("PATH") == "/tmp"
+
+    no_overrides_win = session.windows[0]
+    pane = no_overrides_win.panes[0]
+    pane.send_keys("echo $FOO")
```

Comment:
in re: https://github.com/tmux-python/tmuxp/pull/845#issuecomment-1320908658

I needed this locally for the tests to pass

```suggestion
    pane = no_overrides_win.panes[0]
    time.sleep(0.3)  # To let shells with slow startup catch up
    pane.send_keys("echo $FOO")
```

Reply:
That positions seems a bit arbitrary for the sleep. How about waiting after the full session is built (line 346)?

Reply:
@zappolowski Can you push a change so I can see what you mean?

Reply:
> That positions seems a bit arbitrary for the sleep. How about waiting after the full session is built (line 346)?

I think I get what you mean (though why not just commit it?) [I can't commit it](https://github.com/tmux-python/tmuxp/pull/845#issuecomment-1321236255).

Reply:
Confirmed on my side this works:

```diff
diff --git a/tests/workspace/test_builder.py b/tests/workspace/test_builder.py
index 2852c2f2..98463f9a 100644
--- a/tests/workspace/test_builder.py
+++ b/tests/workspace/test_builder.py
@@ -344,6 +344,7 @@ def test_environment_variables(session):

     builder = WorkspaceBuilder(sconf=workspace)
     builder.build(session)
+    time.sleep(0.3)

     assert session.getenv("FOO") == "SESSION"
     assert session.getenv("PATH") == "/tmp"
```

Reply:
Well done!